src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,57 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!instances.containsKey(clazz)) {
+            try {
+                Constructor<?> constructor = clazz.getConstructor();
+                Object instance = constructor.newInstance();
+                instances.put(clazz, instance);
+                for (Field field : clazz.getDeclaredFields()) {
+                    if (field.isAnnotationPresent(Inject.class)) {
+                        Object fieldInstance = getInstance(field.getType());
+                        field.setAccessible(true);
+                        field.set(instance, fieldInstance);
+                    }
+                }
+            } catch (NoSuchMethodException | IllegalAccessException
+                     | InstantiationException
+                     | InvocationTargetException e) {
+                throw new RuntimeException("Failed to instantiate class: " + clazz.getName(), e);
+            }
+        }
+        return instances.get(clazz);
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
```

Comment:
Chacklist:
"Make Interface Implementations map a class field. You can fill it in using Map.of()."